src/lib/mina_net2/mina_net2.ml

```diff
@@ -1147,7 +1147,9 @@ module Multiaddr = struct
 
   let to_peer t =
     match String.split ~on:'/' t with
-    | [""; "ip4"; ip4_str; "tcp"; tcp_str; "p2p"; peer_id] -> (
+    | [""; protocol; ip4_str; "tcp"; tcp_str; "p2p"; peer_id]
+      when List.mem ["ip4"; "ip6"; "dns4"; "dns6"] protocol ~equal:String.equal
```

Comment:
This will allow `dns4` as a protocol, but the parsing below will fail because the domain name isn't an IPv4 address, so there will still be an error.